### PR TITLE
feat: Add 404 page

### LINKED ISSUE #31: Add 404 to catch broken links
Ensure that all new docs are published to updated URLs and add a 404 which can add a banner to redirected pages indicating the version bump.

### PR DESCRIPTION
**Pull request by mbssantos**
fixes #31

### CODE REVIEW

File: site/src/404.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+var regions = d3.set(data.map(function(d) { return d.type; }));
+var color = d3.scaleOrdinal(d3.schemeCategory10)
+  .domain(regions.values());
+
+var legend = d3.legendColor().scale(color);
+
+var pointSeries = fc.seriesSvgPoint()
+    .crossValue(function(d) { return d.x; })
+    .mainValue(function(d) { return d.y; })
+    .size($('#chart-404').width() <= 768 ? 500 : 2500)
```
Comment (mbssantos) [inline site/src/404.js]: Wasn't sure how to change size based on window width but didn't want to lose too much time on it either

**Comment by mbssantos** [comment]
Desktop:
desktop404

Mobile:
mobile404

File: site/src/404.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    right: 0;
+}
+
+svg.legend {
+    width: 50px;
+    height: 50px;
+}
+
+#chart-404 text {
+	fill: white;
```
Comment (ColinEberhardt) [inline site/src/404.css]: you've got mixed tabs & spaces here. Can you please use 2-space indentation?

File: site/src/404.js (hunk 1 above)
Comment (ColinEberhardt) [inline site/src/404.js]: I don't think there is an easy way - you're approach looks good to me :-)

File: site/src/404.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+var chart = fc.chartSvgCartesian(
+              d3.scaleLinear(),
+              d3.scaleLinear()
+            )
+    .xDomain([0, 13])
+    .yDomain([0, 13])
+    .yOrient('left')
+    .plotArea(pointSeries)
+    .decorate(function(selection) {
```
Comment (ColinEberhardt) [inline site/src/404.js]: I'm not sure the legend really adds much value to this chart? I'd remove it and keep it nice and simple.

File: site/src/404.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+layout: default
+section: introduction
+title: 404 - Page not found
+permalink: /404.html
```
Comment (ColinEberhardt) [inline site/src/404.md]: I'm not sure `permalink` does anything? This isn't using Jekyll

File: site/src/404.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (ColinEberhardt) [inline site/src/404.md]: ^^^ this files isn't markdown, so I'd just call it `404.html`

File: site/src/404.md (hunk 4 above)
Comment (mbssantos) [inline site/src/404.md]: Ah sorry - talking with @chrisprice yesterday I thought we weren't going to use Jekyll?

**Comment by mbssantos** [comment]
@ColinEberhardt All sorted

**Comment by ColinEberhardt** [comment]
👍

**Comment by ColinEberhardt** [comment]
@mbssantos looks good to me. Any chance you could rebase and resolve the merge issue?

**Comment by mbssantos** [comment]
@ColinEberhardt sorted the conflict with the lint PR

**Pull request merged**